### DISCUSSION TITLE
little script to make do_tts.py a tiny gui

### DISCUSSION THREAD

**Discussion by gsgoldma** [neonbjb/tortoise-tts#290] [General]
dont know if this is stupid or anyone will find this useful but i used chatgpt to make this little script

```
import subprocess
import tkinter as tk

def generate_audio(text, voice):
    subprocess.call(["conda", "activate", "voice"], shell=True)
    subprocess.call(["python", "d:\\speak\\tortoise\\do_tts.py", "--text", text, "--voice", voice, "--preset", "fast"], shell=True)

root = tk.Tk()
root.title("TTS Generator")

text_label = tk.Label(root, text="Text:")
text_entry = tk.Entry(root)

voice_label = tk.Label(root, text="Voice:")
voice_entry = tk.Entry(root)

generate_button = tk.Button(root, text="Generate", command=lambda: generate_audio(text_entry.get(), voice_entry.get()))

text_label.grid(row=0, column=0, padx=10, pady=10)
text_entry.grid(row=0, column=1, padx=10, pady=10)

voice_label.grid(row=1, column=0, padx=10, pady=10)
voice_entry.grid(row=1, column=1, padx=10, pady=10)

generate_button.grid(row=2, column=1, padx=10, pady=10, sticky="E")

root.mainloop()
```

just put this python script in the tortoise folder and if activated will make a little text input area for voice/sentence to avoid command line stuff everytime you want to try to run it.

you can also use this bat file to automatically activate your conda env and start the above script just with a click
```
@echo OFF
rem How to run a Python script in a given conda environment from a batch file.

rem It doesn't require:
rem - conda to be in the PATH
rem - cmd.exe to be initialized with conda init

rem Define here the path to your conda installation
set CONDAPATH=<your conda path here>
rem Define here the name of the environment
set ENVNAME=<your env here>

rem The following command activates the base environment.
rem call D:\Miniconda3\Scripts\activate.bat D:\Miniconda3
if %ENVNAME%==base (set ENVPATH=%CONDAPATH%) else (set ENVPATH=%CONDAPATH%\envs\%ENVNAME%)

rem Activate the conda environment
rem Using call is required here, see: https://stackoverflow.com/questions/24678144/conda-environments-and-bat-files
call %CONDAPATH%\Scripts\activate.bat %ENVPATH%

rem Run a python script in that environment
python <your  gui.py>

```
if you make the batfile into a shortcut, you can put cmd /c in front of the path in properties and you can pin it on your taskbar then

**Comment by drgutman** [comment]
I've been writing an interface for TorToiSe for a couple of weeks, but being my first project, I'm learning python as I go along. It will have a small text editor and all the switches that TTS has, plus an option to rearrange the lines of the text to be n characters long, so it won't throw so many errors when you read a long text file. Oh, and it will select between do_tts.py and read.py based on the length of the text. 
I want to add many more features in the future (audio player with seek bar, mp3 export, pdf to txt so it can read pdf's too) but for now I'll stick to the default ones, tho' I'm stuck with some dumb tkinter issues. I hope to upload a version in the next few weeks.

**Comment by wanggilbert** [comment]
I am trying to the same here ,but  it is still not working, can you help me with it?
import os
import tkinter as tk
from tkinter import filedialog, messagebox
from tortoise.api import TextToSpeech
from tortoise.utils.audio import load_audio


class TextToSpeechGUI:
    def __init__(self):
        self.window = tk.Tk()
        self.window.title("Text To Speech")
        
        # Add label for instructions
        label = tk.Label(self.window, text="Select a text file to read:")
        label.pack(pady=10)
        
        # Add button to select file
        file_button = tk.Button(self.window, text="Select File", command=self.select_file)
        file_button.pack()
        
        # Get a list of voice folder names
        voices_dir = r'C:\AI\Voice\tortoise-tts\tortoise\voices'
        voice_folders = [name for name in os.listdir(voices_dir) if os.path.isdir(os.path.join(voices_dir, name))]
        
        # Add dropdown menu for voice selection
        self.voice_var = tk.StringVar(self.window)
        self.voice_var.set("Choose a voice")
        voice_dropdown = tk.OptionMenu(self.window, self.voice_var, *voice_folders)
        voice_dropdown.pack(pady=10)
        
        # Add text widget for user input
        self.text_entry = tk.Text(self.window, height=10, width=50)
        self.text_entry.pack(pady=10)
        
        # Add button to read text
        read_button = tk.Button(self.window, text="Read Text", command=self.read_text)
        read_button.pack(pady=10)
        
        # Initialize TextToSpeech object
        self.tts = TextToSpeech()
        self.reference_clips = []
        
    def select_file(self):
        # Use filedialog to select a file
        file_path = filedialog.askopenfilename(title="Select a text file", filetypes=[("Text files", "*.txt")])
        if file_path:
            with open(file_path, "r") as f:
                self.text_entry.insert(tk.END, f.read())
        
    async def read_text(self):
        text = self.text_entry.get("1.0", "end-1c")
        if not text:
            messagebox.showwarning("Error", "Please enter some text to read")
            return
        try:
            voice_folder = self.voice_var.get()
            voices_dir = r'C:\AI\Voice\tortoise-tts\tortoise\voices'
            voice_path = os.path.join(voices_dir, voice_folder)
            clips_paths = [os.path.join(voice_path, f) for f in os.listdir(voice_path) if f.endswith('.wav')]
            self.reference_clips = [load_audio(os.path.join(voice_path, clip), 22050) for clip in clips_paths]
            await self.tts.speak(text, reference_clips=self.reference_clips)
        except Exception as e:
            messagebox.showerror("Error", str(e))


    def run(self):
        self.window.mainloop()

if __name__ == "__main__":
    app = TextToSpeechGUI()
    app.run()

**Comment by fakerybakery** [comment]
Has anyone considered using Gradio? IMHO the interface is much more intuitive than Tk.
